scripts/SREP/cluster-operator-status/script.sh

```diff
@@ -0,0 +1,137 @@
+print_header() {
+    clusteroperator=$1
+    heading=$2
+    echo -e "--- [ CLUSTER OPERATOR '${clusteroperator}' ${heading} ] ---\n"
+}
+
+# get_summary prints the basic status for the given cluster operator parameter
+get_summary() {
+    clusteroperator=$1
+
+    # shellcheck disable=SC2016
+    available=$(jq -r --arg CLUSTER_OPERATOR "${clusteroperator}" '.items[] | select(.metadata.name == $CLUSTER_OPERATOR) | .status.conditions[] | select(.type == "Available") | "\(.status) --> \(.message)"' <<< "$OPERATOR_REPORT")
+    # shellcheck disable=SC2016
+    progressing=$(jq -r --arg CLUSTER_OPERATOR "${clusteroperator}" '.items[] | select(.metadata.name == $CLUSTER_OPERATOR) | .status.conditions[] | select(.type == "Progressing") | "\(.status) --> \(.message)"' <<< "$OPERATOR_REPORT")
+    # shellcheck disable=SC2016
+    degraded=$(jq -r --arg CLUSTER_OPERATOR "${clusteroperator}" '.items[] | select(.metadata.name == $CLUSTER_OPERATOR) | .status.conditions[] | select(.type == "Degraded") | "\(.status) --> \(.message)"' <<< "$OPERATOR_REPORT")
+
+    print_header "${clusteroperator}" "SUMMARY"
+    echo " - AVAILABLE:   ${available}"
+    echo " - DEGRADED:    ${degraded}"
+    echo " - PROGRESSING: ${progressing}"
+    echo -e "\n"
+}
+
+# get_description prints a detailed status description for the given cluster 
+# operator parameter
+get_description() {
+    clusteroperator=$1
+
+    if ! co_desc_report=$(oc describe clusteroperator "${clusteroperator}"); then
+       echo "An error was returned when describing ClusterOperator '${clusteroperator}': ${co_desc_report}"
+       return
+    fi
+    print_header "${clusteroperator}" "DESCRIPTION"
+    echo -e "${co_desc_report}\n\n"
+}
+
+# get_logs prints the last 15 lines of logs from the cluster operator pod
+get_logs() {
+    clusteroperator=$1
+    # Look up all namespaces referenced in status
+    # shellcheck disable=SC2016
+    relatedns=$(jq -r --arg CLUSTER_OPERATOR "${clusteroperator}" '.items[] | select(.metadata.name == $CLUSTER_OPERATOR) | .status.relatedObjects[] | select(.resource == "namespaces") | .name' <<< "$OPERATOR_REPORT")
+    while IFS= read -r ns; do
+        ns_report=$(oc get deployment --ignore-not-found -n "$ns" -o name)
+        while IFS= read -r deployment; do
+            # Look for any deployment which matches an operator deployment.
+            # The cluster operator name may or may not have the word operator in it.
+            # The cluster operator namespaces may or may not unrelated operator pods in them.
+            if [[ "${deployment}" =~ ${clusteroperator} && "${deployment}${clusteroperator}" =~ operator ]]; then
+                # deploymentlogs=$(log_cmd "${deployment}" "${ns}")
+                if [[ -n "${LOG_PATTERN}" ]]; then
+                    # shellcheck disable=SC2046
+                    deploymentlogs=$(grep -E "${LOG_PATTERN}" <<< $(oc logs "${deployment}" -n "${ns}" --all-containers "${OC_LOGS_SINCE}") || true)
```

Comment:
```suggestion
                    deploymentlogs=$(grep -iE "${LOG_PATTERN}" <<< $(oc logs "${deployment}" -n "${ns}" --all-containers "${OC_LOGS_SINCE}") || true)
```
Using `-E` might be more strict in having exact patterns to be matched where the user is expected to know the error strings. Would it make sense to accept strings by ignoring case? Else another option is to have `GREP_FLAGS` env variable but don't want to add too many flags and knobs for users in the start. Am ok with `-E` too but then we could document the defaults so that the user is aware that the strings are case sensitive.

Reply:
I've changed it to `-iE`